db/dcrpg/queries.go

```diff
@@ -4121,6 +4121,30 @@ func RetrieveBlockStatus(ctx context.Context, db *sql.DB, hash string) (bs dbtyp
 	return
 }
 
+// RetrieveBlockStatuses retrieves the block chain statuses of all blocks at
+// the given height.
+func RetrieveBlockStatuses(ctx context.Context, db *sql.DB, idx int64) (blocks []*dbtypes.BlockStatus, err error) {
+	var rows *sql.Rows
+	rows, err = db.QueryContext(ctx, internal.SelectBlockStatuses, idx)
+	if err != nil {
+		return
+	}
+	defer closeRows(rows)
+
+	for rows.Next() {
+		var bs dbtypes.BlockStatus
+		err = rows.Scan(&bs.IsValid, &bs.IsMainchain, &bs.Height, &bs.Hash)
```

Comment:
Since height from `SelectBlockStatuses` is necessarily equal to `idx`, I'd leave that out of the query and just set `bs.Height = idx` after a error-free `Scan`.